L1Trigger/TrackerTFP/interface/DataFormats.h

```diff
@@ -0,0 +1,672 @@
+      {}                                                                                                                                                                    // Process::dr
+    }};
+    // track word assembly, shows which track variables are used by each process
+    static constexpr std::array<std::initializer_list<Variable>, +Process::end> tracks_ = {{
+      {},                                                                                                                                                                                            // Process::fe
+      {},                                                                                                                                                                                            // Process::dtc
+      {},                                                                                                                                                                                            // Process::pp
+      {},                                                                                                                                                                                            // Process::gp
+      {},                                                                                                                                                                                            // Process::ht
+      {},                                                                                                                                                                                            // Process::mht
+      {},                                                                                                                                                                                            // Process::sf
+      {Variable::hitPattern, Variable::layerMap, Variable::hitPattern, Variable::phiT, Variable::qOverPt, Variable::zT, Variable::cot, Variable::sectorPhi, Variable::sectorEta, Variable::trackId}, // Process::kfin
+      {Variable::phiT, Variable::qOverPt, Variable::zT, Variable::cot, Variable::sectorPhi, Variable::sectorEta, Variable::match},                                                                   // Process::kf
+      {Variable::phi0, Variable::qOverPt, Variable::z0, Variable::cot}                                                                                                                               // Process::dr
+    }};
+  public:
+    DataFormats();
+    DataFormats(const edm::ParameterSet& iConfig, const trackerDTC::Setup* setup);
+    ~DataFormats(){}
+    // converts bits to ntuple of variables
+    template<typename ...Ts>
+    void convertStub(const TTDTC::BV& bv, std::tuple<Ts...>& data, Process p) const;
```

Comment:
In general, put input arguments before output arguments. It makes it easier to understand which arguments a function is producing.

Reply:
done

Reply:
In convertStub, isn't "Process p" an input argument, but it appears last?

Reply:
ah I didn't consider config as input an wrote the order input, output, config. Now I understand your problem although I find mine style easier to understand I will change it.